procfile/procfile.go

```diff
@@ -57,3 +65,51 @@ func NewProcfileFromPath(path string) (Procfile, error) {
 
 	return p, nil
 }
+
+// NewProcfileFromBinding creates a Procfile by reading Procfile from bindings if it exists.  If it does not exist, returns an
+// empty Procfile.
+func NewProcfileFromBinding(binds libcnb.Bindings) (Procfile, error) {
+
+	p := Procfile{}
+	if binding, ok, err := bindings.ResolveOne(binds, bindings.OfType(BindingType)); err != nil {
+		return nil, fmt.Errorf("unable to resolve binding\n%w", err)
+	} else if ok {
+		if path, ok := binding.SecretFilePath(BindingType); ok {
+			if p, err = NewProcfileFromPath(filepath.Dir(path)); err != nil {
+				return nil, err
+			}
+			return p, nil
+		} else {
+			return nil, fmt.Errorf("unable to find Procfile from binding")
+		}
+	} else {
+		return p, nil
+	}
+}
+
+// NewProcfileFromPathOrBinding attempts to create a merged Procfile from given path and bindings.  If neither can be created, returns an
+// empty Procfile.
+func NewProcfileFromPathOrBinding(path string, binds libcnb.Bindings) (Procfile, error) {
+
+	procPath, err := NewProcfileFromPath(path)
+	if err != nil {
+		return nil, err
+	}
+	procBind, err := NewProcfileFromBinding(binds)
+	if err != nil {
+		return nil, err
+	}
+	procBind = mergeProcfiles(procPath, procBind)
```

Comment:
FYI only. You could return this directly.

```
return mergeProcfiles(procPath, procBind), nil
```

Reply:
😨